### PR TITLE
Add typings for ioredis-mock

### PR DESCRIPTION
**Pull request by lukaselmer**
Typings for https://www.npmjs.com/package/ioredis-mock

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] (based on redis-mock): Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] (needed additional rule `NeedsExportEquals`) `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/ioredis-mock/tslint.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+    "rules": {
+        "npm-naming": [
+            true,
+            {
+                "mode": "code",
+                "errors": [["NeedsExportEquals", false]]
+            }
+        ]
+    }
+}
```
Comment (nicolas377) [inline types/ioredis-mock/tslint.json:12]: The rules needs to be removed (see [here](https://github.com/DefinitelyTyped/DefinitelyTyped#user-content-linter-tslintjson))

```suggestion
{
    "extends": "dtslint/dt.json"
}
```

**Review by nicolas377 — CHANGES_REQUESTED** [review]

File: types/ioredis-mock/tslint.json (hunk 1 above)
Comment (lukaselmer) [reply types/ioredis-mock/tslint.json:12]: @nicolas377 I tried making it work without this rule, but didn't manage to do it. Is it even possible? I noticed many other typings which also seem to need this rule. The command line told me to add this rule if I think this error is wrong:

image

**Review by nicolas377 — APPROVED** [review]
LGTM!

**Comment by lukaselmer** [comment]
Ready to merge

**Comment by lukaselmer** [comment]
@orta do you have time for a review? thank you!

**Review by orta — APPROVED** [review]
Cool, looks good to me 👍🏻 

Nice work on the tests

**Comment by lukaselmer** [comment]
> Cool, looks good to me 👍🏻
> 
> Nice work on the tests

Thank you =)

**Comment by lukaselmer** [comment]
Ready to merge

**Pull request merged**